Functionality related to datatype ShareDocumentsRequest (#172)
digipost/digipost-api-client-java: docs/_v15_x/2_send.md

        .send();
```

### Datatype ShareDocumentsRequest

> This datatypes facilitates exchange of documents between an organisation and a Digipost end user. The sender

Reviewer: ```suggestion
This datatype can be used by organisations to share documents with an Digipost end user. The sender
```
"facilitate", too vague?
"exchange": too general? It is a one-way sharing process, not a mutual exchange- Exchange can even be misunderstood as _messaging_, in general.

first sends a message of datatype ShareDocumentsRequest, to which the end user can attach a list of documents. When
new documents are shared, a DocumentEvent is generated. The organisation can retrieve the status of their
ShareDocumentsRequest. If documents are shared and the sharing is not cancelled, the documents can either be downloaded
or viewed on the digipostdata.no domain. Active requests can be cancelled both by the end user and the organisation.

The `purpose` attribute of the ShareDocumentsRequest should briefly explain why the sender organisation want to gain

> access to the relevant documents. The primary document should contain a more detailed explanation.

Reviewer 2: Mulig vi burde vektlegge enda mer at dette er den juridiske grunnen til at virksomheten ønsker å få innsyn i dokumentet. Kanskje vår personvern-ekspert eller sikkerhetsansvarlig har en god formulering på lur? @sarawe @hanskhe

Reviewer 3: Kanskje legge til noe sånn som: "This text will be displayed prominently, and should contain the information necessary for the user to make an informed choice." før den siste setningen?

#### Send ShareDocumentsRequest
```java
PersonalIdentificationNumber pin = new PersonalIdentificationNumber("26079833787");
UUID messageUUID = UUID.randomUUID();

ShareDocumentsRequest shareDocumentsRequest = new ShareDocumentsRequest(
        Duration.ofDays(60).toSeconds(),
        "We require to see your six latest pay slips in order to give you a loan."
);

Document primaryDocument = new Document(messageUUID, "Request to access your latest payslips", FileType.PDF, shareDocumentsRequest);

Message message = Message.newMessage("messageId", primaryDocument)
        .recipient(pin)
        .build();

client.createMessage(message)
        .addContent(primaryDocument, Files.newInputStream(Path.of("longer-desc-of-sharing-purpose.pdf")))
        .send();
```

> #### Be notified of new shared documents

Reviewer 2: Småpirk, men lurte på om vi ikke burde bruke ordet "notified" siden de faktisk må polle selv. Men helt opp til deg om du vil endre på det eller ikke 😊
